filament/backend/src/opengl/OpenGLDriver.cpp

```diff
@@ -3023,6 +3023,16 @@ void OpenGLDriver::draw(PipelineState state, Handle<HwRenderPrimitive> rph) {
     DEBUG_MARKER()
 
     OpenGLProgram* p = handle_cast<OpenGLProgram*>(state.program);
+
+    // If the material debugger is enabled, avoid fatal (or cascading) errors and that can occur
+    // during the draw call when the program is invalid. The shader compile error has already been
+    // dumped to the console at this point, so it's fine to simply return early.
+    #if FILAMENT_ENABLE_MATDBG
+    if (UTILS_UNLIKELY(!p->isValid())) {
```

Comment:
I think you can even write `if (FILAMENT_ENABLE_MATDBG && ...)` here, this way these lines always get compiled (but thrown out by the optimizer) which helps making sure we don't break the code when it's not enabled.